zap/src/test/java/org/parosproxy/paros/core/scanner/VariantPlainBodyUnitTest.java

```diff
@@ -0,0 +1,116 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.parosproxy.paros.core.scanner;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.emptyString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+
+/** Unit test for {@link VariantPlainBody}. */
+class VariantPlainBodyUnitTest {
+
+    @Test
+    void shouldHaveParametersListEmptyByDefault() {
+        // Given
+        VariantPlainBody variant = new VariantPlainBody();
+        // When
+        List<NameValuePair> parameters = variant.getParamList();
+        // Then
+        assertThat(parameters, is(empty()));
+    }
+
+    @Test
+    void shouldNotAllowToModifyReturnedParametersList() {
+        // Given
+        VariantPlainBody variant = new VariantPlainBody();
+        NameValuePair nvp = new NameValuePair(NameValuePair.TYPE_POST_DATA, "foo", "bar", 0);
+        // When / Then
+        assertThrows(UnsupportedOperationException.class, () -> variant.getParamList().add(nvp));
+    }
+
+    @Test
+    void shouldFailToExtractParametersFromUndefinedMessage() {
+        // Given
+        VariantPlainBody variant = new VariantPlainBody();
+        HttpMessage undefinedMessage = null;
+        // When / Then
+        assertThrows(IllegalArgumentException.class, () -> variant.setMessage(undefinedMessage));
+    }
+
+    @Test
+    void shouldNotExtractAnyParameterIfThereIsNoBodyContent() {
+        // Given
+        VariantPlainBody variant = new VariantPlainBody();
+        HttpMessage msg = createBasicMessage();
+        // When
+        variant.setMessage(msg);
+        // Then
+        assertThat(variant.getParamList(), is(empty()));
+    }
+
+    @Test
+    void shouldNotExtractAnyParameterIfThereIsAContentType() {
+        // Given
+        VariantPlainBody variant = new VariantPlainBody();
+        HttpMessage msg = createBasicMessage();
+        msg.getRequestHeader().setHeader(HttpHeader.CONTENT_TYPE, "application/foo");
+        // When
+        variant.setMessage(msg);
+        // Then
+        assertThat(variant.getParamList(), is(empty()));
+    }
+
+    @Test
+    void shouldExtractParameterIfThereIsBodyContent() {
```

Comment:
`…AndNoContentType` and there should be a test with text plain and body still produces a NVP.

Reply:
I don't think the second part was addressed.

Reply:
Now testing both missing (set null) and text/plain

Reply:
The name of the test should be updated to match the latest changes. The use of `ValueSource` and `NullSource` would be simpler/clearer (IMO, it's not making use of CSVs).

Reply:
Tweaked

Reply:
Was it meant `NoOr`?

Reply:
Yes, Fixed